lightning/src/ln/mod.rs

```diff
@@ -35,12 +35,18 @@ mod channel;
 mod onion_utils;
 mod wire;
 
+// Older rustc (which we support) refuses to let us call the get_payment_preimage_hash!() macro
+// without the node parameter being mut. This is incorrect, and thus newer rustcs will complain
+// about an unnecessary mut. Thus, we silence the unused_mut warning in two test modules below.
+
 #[cfg(test)]
 #[macro_use]
 pub(crate) mod functional_test_utils;
 #[cfg(test)]
+#[allow(unused_mut)]
 mod functional_tests;
 #[cfg(test)]
+#[allow(unused_mut)]
```

Comment:
Hmm, tests are passing for me without these lines on 1.30.0 and 1.34.2. Which version was this failing on for you?

Reply:
Tests dont fail anywhere, but without this you get about a mile of "warning: you dont need a mut here" things when building on rustc 1.46.